doc/manual/source/development/documentation.md

```diff
@@ -25,20 +25,31 @@ nix build .#nix-manual
 and open `./result/share/doc/nix/manual/index.html`.
 
 
-To build the manual incrementally, [enter the development shell](./building.md) and run:
+To build the manual incrementally, [enter the development shell](./building.md) and configure with `doc-gen` enabled:
+
+**If using interactive `nix develop`:**
 
 ```console
-make manual-html-open -j $NIX_BUILD_CORES
+$ nix develop
+$ mesonFlags="$mesonFlags -Ddoc-gen=true" mesonConfigurePhase
 ```
 
-In order to reflect changes to the [Makefile for the manual], clear all generated files before re-building:
+**If using direnv:**
 
-[Makefile for the manual]: https://github.com/NixOS/nix/blob/master/doc/manual/local.mk
+```console
+$ direnv allow
+$ bash -c 'source $stdenv/setup && mesonFlags="$mesonFlags -Ddoc-gen=true" mesonConfigurePhase'
```

Comment:
Can't be done with `nix develop --configure` or `nix develop --command`; I've tried.
I won't open an issue, because I don't think `nix develop --configure` should exist, and the fix to `--command` is to let it be controlled by Nixpkgs (https://github.com/NixOS/nix/issues/7501)